plugin/keyring_vault/vault_keyring.cc

```diff
@@ -0,0 +1,198 @@
+#include <my_global.h>
+#include <mysql/plugin_keyring.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include "keyring.h"
+#include "vault_keys_container.h"
+#include "vault_parser.h"
+#include "vault_io.h"
+
+using keyring::IVault_curl;
+using keyring::IVault_parser;
+using keyring::Vault_parser;
+using keyring::Vault_io;
+using keyring::Vault_keys_container;
+using keyring::Vault_curl;
+using keyring::Logger;
+
+int check_keyring_file_data(MYSQL_THD thd  MY_ATTRIBUTE((unused)),
+                            struct st_mysql_sys_var *var  MY_ATTRIBUTE((unused)),
+                            void *save, st_mysql_value *value)
+{
+  char            buff[FN_REFLEN+1];
+  const char      *keyring_filename;
+  int             len = sizeof(buff);
+  boost::movelib::unique_ptr<IKeys_container> new_keys(new Vault_keys_container(logger.get()));
```

Comment:
no std::unique_ptr - due to no c++11. keys container is kept in keys variable which is of type unique_ptr<IKeys_contrainer> defined in plugin/keyring/common/keyring_impl.cc. Story behind this is that boost was used more extensively, but due to "politic decision" different parts of boost were taken out of keyring.